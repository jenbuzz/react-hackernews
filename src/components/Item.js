import React from 'react';
import PropTypes from 'prop-types';
import dayjs from 'dayjs';

const Item = ({ item }) => {
    return item && item.url ? (
        <div>
            <a href={item.url} target="_blank" rel="noopener noreferrer nofollow">
                {item.title}
            </a>
            {item.time ? (
                <div>
                    <span>{dayjs.unix(item.time).format('YYYY-MM-DD')}</span>
                </div>
            ) : null}
        </div>
    ) : null;
};

Item.propTypes = {
    item: PropTypes.object,
};

export default Item;
