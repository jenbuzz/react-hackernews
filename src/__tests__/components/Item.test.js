import React from 'react';
import { shallow } from 'enzyme';
import Item from '../../components/Item';

describe('Item', () => {
    it('renders - item not set', () => {
        const component = shallow(<Item />);

        expect(component).toMatchSnapshot('item-not-set');
    });

    it('renders - item is set', () => {
        const item = {
            title: 'Lorem ipsum',
            url: 'http://localhost',
            time: 1591024073,
        };

        const component = shallow(<Item item={item} />);

        expect(component).toMatchSnapshot('item-is-set');
    });
});
