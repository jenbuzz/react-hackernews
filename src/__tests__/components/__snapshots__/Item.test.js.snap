// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders - item is set: item-is-set 1`] = `
<div>
  <a
    href="http://localhost"
    rel="noopener noreferrer nofollow"
    target="_blank"
  >
    Lorem ipsum
  </a>
  <div>
    <span>
      2020-06-01
    </span>
  </div>
</div>
`;

exports[`Item renders - item not set: item-not-set 1`] = `""`;
